packages/stores/src/stores/createStore.ts

```diff
@@ -81,9 +81,13 @@ export const createStore = <T extends { [key: string]: any }>(
             }
         });
         const unReset = on('reset', () => cb(defaultState?.[propName] as any));
+        const unDelete = on('delete', () =>
+            cb(defaultState?.[propName] as any),
```

Comment:
Also, this will call `onChange` for  any key that's deleted. e.g.

```ts
const store = createStore({ a: true, b: 3 });

store.onChange("a", () => console.log("a changed"));

store.delete("b");

// console logs "a changed"
```

You should be checking the key that is passed in the `on` callback. Something like:

```ts
const unDelete = on('delete', (key) => {
   if (key !== propName) return;
   cb(backingMap.get(propName)),
 })
 ```

Add a test to make sure it _isn't_ called on other prop deletions, for both.